src/hooks/useWallpaper.jsx

```diff
@@ -18,46 +23,79 @@ const useWallpaper = () => {
       }
       try {
         setFetchStatus('loading')
-        const binary = await localForage.getItem('customWallpaper')
-        if (binary) {
-          setBinaryCustomWallpaper(binary)
-        }
-        const response = await client
-          .collection('io.cozy.files')
-          .getDownloadLinkByPath(homeConfig.customWallpaperPath)
-        setWallpaperLink(response)
-        setFetchStatus('loaded')
-        const fetchBinary = await fetch(response)
-        const blob = await fetchBinary.blob()
-        const reader = new FileReader()
-        reader.readAsDataURL(blob)
-        reader.onloadend = async () => {
-          const base64data = reader.result
-          setBinaryCustomWallpaper(base64data)
-          await localForage.setItem('customWallpaper', base64data)
-          localStorage.setItem('hasCustomWallpaper', true)
+        const link = await localForage.getItem('customWallpaper')
+        if (link) {
+          setWallpaperLink(link)
+          setFetchStatus('loaded')
+          return
         }
       } catch (error) {
         await localForage.removeItem('customWallpaper')
         localStorage.setItem('hasCustomWallpaper', false)
         setWallpaperLink(cozyDefaultWallpaper)
-        setBinaryCustomWallpaper(null)
         setFetchStatus('failed')
       }
     }
     fetchData()
   }, [client, cozyDefaultWallpaper])
 
+  const setWallpaperLinkAndStore = async link => {
+    setWallpaperLink(link)
+    if (link === cozyDefaultWallpaper) {
+      await localForage.removeItem('customWallpaper')
+      localStorage.setItem('hasCustomWallpaper', false)
+    }
+    localStorage.setItem('hasCustomWallpaper', true)
```

Comment:
```suggestion
      await localForage.removeItem('customWallpaper')
      localStorage.setItem('hasCustomWallpaper', false)
      
      return
    }
    localStorage.setItem('hasCustomWallpaper', true)
```

This is redundant. We set false in the local storage and then true. i think a return is missing here

Reply:
Good catch thanks. I fixed it with a else which looks clearer to me here.